frontend/src/components/UserSettings/UserNamingFormat.spec.js

```diff
@@ -0,0 +1,86 @@
+import { mount } from '@vue/test-utils'
+import UserNamingFormat from './UserNamingFormat.vue'
+import { toastErrorSpy } from '@test/testSetup'
+
+const mockAPIcall = jest.fn()
+
+const storeCommitMock = jest.fn()
+
+const localVue = global.localVue
+
+describe('UserNamingFormat', () => {
+  let wrapper
+  beforeEach(() => {
+    wrapper = mount(UserNamingFormat, {
+      mocks: {
+        $t: (key) => key, // Mocking the translation function
+        $store: {
+          state: {
+            gmsPublishName: null,
+          },
+          commit: storeCommitMock,
+        },
+        $apollo: {
+          mutate: mockAPIcall,
+        },
+      },
+      localVue,
+      propsData: {
+        selectedOption: 'PUBLISH_NAME_NONE',
+        initialValue: 'PUBLISH_NAME_NONE',
+        attrName: 'publishName',
+        successMessage: 'success message',
+      },
+    })
+  })
+
+  afterEach(() => {
+    wrapper.destroy()
+  })
+
+  it('renders the correct dropdown options', () => {
+    const dropdownItems = wrapper.findAll('.dropdown-item')
+    expect(dropdownItems.length).toBe(7)
+
+    const labels = dropdownItems.wrappers.map((item) => item.text())
+    expect(labels).toEqual([
+      'settings.publish-name.none',
+      'settings.publish-name.initials',
+      'settings.publish-name.first',
+      'settings.publish-name.first-initial',
+      'settings.publish-name.last',
+      'settings.publish-name.last-initial',
+      'settings.publish-name.full',
```

Comment:
Wie in der EnumKlasse schon angemerkt, sind die GMS-PublishNames-Types wohl unterschiedlich zu den HumHub-Types.
Es fehlt die Variante "optional Alias oder Initials" und die Varianten "Last" und "Last-Initial" gibt es bei GMS nicht.

Reply:
Ach so, dann habe ich das bisher falsch verstanden. Sorry. Ich dachte du hast die Enum-Klasse von den GMSNameTypes verwendet.
Nach der heutigen Entscheidung sollten wir schauen, dass wir für GMS und für HumHub die gleichen Varianten verwenden.